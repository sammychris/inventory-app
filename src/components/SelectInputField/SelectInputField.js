import React from 'react';
import styled from 'styled-components';


const SelectInputField = ({event, bookId, books}) => {
    return (
        <Select value={bookId} onChange={event}>
          <option value="">none</option>
          {
            books.map((item, i) => <option key={i} value={item.id}>{item.book}</option>)
          }
        </Select>
    );
  }


export default SelectInputField;

const Select = styled.select`
  width: 500px;
  height: 30px;
  font-size: 18px;
  margin: 5px 0;
`;
