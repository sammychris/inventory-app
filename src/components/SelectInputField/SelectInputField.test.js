import { shallow } from "enzyme";
import SelectInputField from "./SelectInputField";

test("expect to render a SelectInputField", () => {
  const props = {
    books: [],
    bookId: "0",
    event: jest.fn(),
  };
  const wrapper = shallow(<SelectInputField {...props} />);
  expect(wrapper.length).toEqual(1);
});

test("expect to make a snap shot for SelectInputField component", () => {
  const props = {
    books: [],
    bookId: "0",
    event: jest.fn(),
  };
  const wrapper = shallow(<SelectInputField {...props} />);
  expect(wrapper).toMatchSnapshot();
});