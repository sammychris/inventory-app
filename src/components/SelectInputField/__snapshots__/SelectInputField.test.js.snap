// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to make a snap shot for SelectInputField component 1`] = `ShallowWrapper {}`;
